apps/meteor/app/api/server/v1/moderation.ts

```diff
@@ -113,6 +161,61 @@ API.v1.addRoute(
 	},
 );
 
+// api endpoint to get user reports of a userid
+
+API.v1.addRoute(
+	'moderation.user.reportsByUserId',
+	{
+		authRequired: true,
+		validateParams: isGetUserReportsParams,
+		permissionsRequired: ['view-moderation-console'],
+	},
+	{
+		async get() {
+			const { userId, selector = '' } = this.queryParams;
+			const { sort } = await this.parseJsonQuery();
+			const { count = 50, offset = 0 } = await getPaginationItems(this.queryParams);
+
+			const user = await Users.findOneById<IUser>(userId, {
+				projection: {
+					_id: 1,
+					username: 1,
+					name: 1,
+					avatarETag: 1,
+					active: 1,
+					roles: 1,
+					emails: 1,
+					createdAt: 1,
+				},
+			});
+
+			const escapedSelector = escapeRegExp(selector);
+			const { cursor, totalCount } = ModerationReports.findUserReportsByReportedUserId(userId, escapedSelector, {
+				offset,
+				count,
+				sort,
+			});
+
+			const reports = await cursor.toArray();
+			const emailSet = new Map<IUserEmail['address'], IUserEmail>();
+
+			reports.flatMap((report) => report.reportedUser?.emails ?? []).forEach((email) => emailSet.set(email.address, email));
+
+			if (user) {
+				user.emails = Array.from(emailSet.values());
+			}
+
+			return API.v1.success({
+				user,
+				reports,
+				count: reports.length,
+				total: totalCount,
```

Comment:
totalCount is a promise iirc.

Reply:
My bad; I will correct this while working on improving the total count. I plan on making it, equal to the total number of distinct users. Currently, it is shown as the total number of reports. But, it's still pending and might change.